pleasew

```diff
@@ -1,99 +1,122 @@
-#!/usr/bin/env bash
+#!/bin/sh
+
+set -e
 set -u
 
-RED="\x1B[31m"
-GREEN="\x1B[32m"
-YELLOW="\x1B[33m"
-RESET="\x1B[0m"
+set -o errexit
+
+RED='\x1B[31m'
+GREEN='\x1B[32m'
+YELLOW='\x1B[33m'
+RESET='\x1B[0m'
 
-DEFAULT_URL_BASE="https://get.please.build"
+DEFAULT_URL_BASE='https://get.please.build'
 
 OS="$(uname)"
 
-if [[ $OS == "Darwin" ]]; then
-    # switch between mac amd64/arm64 
+if [ "${OS}" = 'Darwin' ]; then
+    # switch between mac amd64/arm64
     ARCH="$(uname -m)"
 else
     # default to amd64 on other operating systems
     # because we only build intel binaries
-    ARCH="amd64"
+    ARCH='amd64'
 fi
 
-# Check PLZ_CONFIG_PROFILE or fall back to arguments for a profile.
-PROFILE="${PLZ_CONFIG_PROFILE:-$(sed -E 's/.*--profile[= ]([^ ]+).*/\1/g' <<< "$*")}"
+# Check `PLZ_CONFIG_PROFILE` or fall back to arguments for a profile.
+PROFILE="${PLZ_CONFIG_PROFILE:-$(
+	sed -E 's/.*--profile[= ]([^ ]+).*/\1/g' <<-EOS
+		"${*}"
+	EOS
+)}"
```

Comment:
From my testing, this seems to add a trailing quote to the profile string. Not sure why though.

Reply:
Indeed, I just made a few tests and the use of `sed` here may not be the best choice. I realized it's not just a quoting problem since `sed` returns the string as-is when it doesn't match, for instance `./pleasew query alltargets` will give a profile of `"query alltargets"`.